code/src/components/TodoItem.js

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { todos } from "reducers/todos";
+
+
+export const TodoItem = ({itemIndex}) => {
+  const item = useSelector((store) => store.todos.list.items[itemIndex]);
+  const dispatch = useDispatch();
+
+  const onRemoveClicked = (event) => {
+    dispatch(
+      todos.actions.removeTodo({
+        itemIndex,
+      })
+    );
+  };
+
+  const handleOnChange = (event) => {
+    dispatch(
+      todos.actions.setDone({
+        itemIndex: itemIndex,
```

Comment:
since the key and value are the same its enough to just write `itemIndex` instead of `itemIndex: itemIndex`

Also, be aware that if you manipulate your todolist in a way so that the indexes changes of the todos, you might be changing or deleting the wrong todos here... Try to use an id for each todo instead of index.